src/main/java/org/springframework/data/redis/stream/StreamMessageListenerContainer.java

```diff
@@ -627,6 +639,21 @@ public StreamMessageListenerContainerOptionsBuilder<K, V> pollTimeout(Duration p
 			return this;
 		}
 
+		/**
+		 * Configure a timeout for shutdown container.
+		 *
+		 * @param shutdownTimeout must not be {@literal null} or negative.
+		 * @return {@code this} {@link StreamMessageListenerContainerOptionsBuilder}.
+		 */
+		public StreamMessageListenerContainerOptionsBuilder<K, V> shutdownTimeout(Duration shutdownTimeout) {
+
+			Assert.notNull(shutdownTimeout, "Shutdown timeout must not be null");
+			Assert.isTrue(!shutdownTimeout.isNegative(), "Shutdown timeout must not be negative");
```

Comment:
The assertion condition should be `shutdownTimeout. isNegative()` instead of `!shutdownTimeout. isNegative()`.

Reply:
Thank you for the reviewing. But here is correct, if you replace it with ` shutdownTimeout. isNegative() ` current assertion, that means you're asking for a negative `shutdownTime` parameter.

Reply:
Sorry, I was confused with the negative of negative